p2719.md

```diff
@@ -432,134 +474,157 @@ Updates to [expr.new]{.sref}
+>    first]{.rm} argument is the amount of space requested, and has type `std​::​size_t`.
+>  - [21.2]{.pnum old=""} If the type of the allocated object has new-extended alignment, the next
 >    argument is the type's alignment, and has type `std​::​align_val_t`. If the new-placement syntax
 >    is used, the initializer-clauses in its expression-list are the succeeding arguments. If no
 >    matching function is found then
 >
->    - [20.1 -> 21.2.1]{.pnum} if the allocated object type has new-extended alignment, the alignment
+>    - [21.2.1]{.pnum old=20.1} if the allocated object type has new-extended alignment, the alignment
 >      argument is removed from the argument list;
 >
->    - [20.2 -> 21.2.2]{.pnum} otherwise, an argument that is the type's alignment and has type
->      `std​::​align_val_t` is added into the argument list immediately after the first argument;
+>    - [21.2.2]{.pnum old=20.2} otherwise, an argument that is the type's alignment and has type
+>      `std​::​align_val_t` is added into the argument list immediately after the size argument;
 >
->  - [+ 21.3]{.pnum} and then overload resolution is performed again.
+>    - [21.2.3]{.pnum old=""} and then overload resolution is performed again.
 
-[For Reviewers: end the added inset + new sublist]
+[end the added inset + new sublist]{.draftnote}
 
 ::: add
->  - [21.4]{.pnum} If no matching function has been found then paragraphs [21.3]-[21.4] are repeated
->    starting from an argument list consisting of a single parameter that is the amount of space
->    requested, and has type `std​::​size_t`.
+>  - [21.3]{.pnum} If no matching function was found and the type-identity argument was present in
+>    the argument list, the above overload resolution steps are repeated without the type-identity
+>    argument`.
 :::
 
 > [29]{.pnum} A declaration of a placement deallocation function matches the declaration of a
 > placement allocation function if it has the same number of parameters and, after parameter
-> transformations ([dcl.fct]{.sref}), all parameter types except [the first]{.rm}[the size and
-> object]{.add} are identical.
+> transformations ([dcl.fct]{.unnumbered .sref}), all parameter types except [the first]{.rm}[the
+> size and object]{.add} are identical.
 
 > [30]{.pnum} If a new-expression calls a deallocation function it passes [the type-identity as the
 > first argument if it was passed to the allocation function, and]{.add} the value returned from
-> the allocation function call as the [first argument]{.rm}[argument for the object parameter]{.add}
+> the allocation function call as the [first argument]{.rm}[argument for the address parameter]{.add}
 > of type `void *`.
 
+[The following restriction was not discussed in Wroclaw, but we believe is a useful hazard
+restriction given the lack of any mechanism to otherwise force co-definition of new and delete
+operators]{.draftnote audience=CWG&EWG}
```

Comment:
While I think that this restriction is great, the right way to call it out to EWG is not in a drafting note. We'd need to add a small Design subsection where we discuss that change.

Reply:
- [ ] todo